### PR TITLE
[air output] Update wording to "Logical Resource Usage".

### PR DESCRIPTION
**Pull request by xwjiang2010**
Update wording to "Logical Resource Usage" and remove heap/object memory stuff.

Remove memory_debug_str on head node. Remove runner.debug_str (dangling code path and duplicated logic with what's used in `tune.run` - I don't think people will debug trial_runner by itself these days)

## Related issue number

#33152

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by xwjiang2010** [comment]
bunch of rllib test failures that I don't think relevant.

**Review by krfricke — COMMENTED** [review]
Code looks good, just two questions:

1) Is "Logical Resource Usage" the correct term here? I don't know what it means. What is a logical resource? Maybe "Resources used" or "Assigned resources" or so?
2) Can we make sure this is found in the documentation somewhere?

**Comment by xwjiang2010** [comment]
> Code looks good, just two questions:
> 
> 1. Is "Logical Resource Usage" the correct term here? I don't know what it means. What is a logical resource? Maybe "Resources used" or "Assigned resources" or so?
> 2. Can we make sure this is found in the documentation somewhere?

"Logical Resource" is in contrast to "Physical Resource". @scottsun94 in his user research, encountered people confused about this. Hence the emphasis on "Logical".

"Usage" is also surfaced as preference from user research cc @scottsun94
To me, it's the same as granted/used/assigned etc. 

I have no preference on any of the terms (apart from "Requested" is clearly misleading) - just wants to follow design suggestions here.

Documentation sounds good.

**Comment by xwjiang2010** [comment]
Actually @krfricke I found it here! 
https://docs.ray.io/en/latest/ray-core/scheduling/resources.html#physical-resources-and-logical-resources

**Comment by scottsun94** [comment]
In Ray core, we already have a dedicated ["Resource" doc page](https://docs.ray.io/en/latest/ray-core/scheduling/resources.html#physical-resources-and-logical-resources) and talk about the difference between "logical resource" and "physical resource"

The gap I found is that people using AIR don't really understand the distinction between logical resources and physical hardware utilization. For example, in my chat with relatively new ray users like @woshiyyya, they don't really know that "resource" refers to the logical resource concept purely for scheduling.

In the long term, we probably need to introduce the "ray resource" or "logical resource" concept into AIR's doc because people shouldn't have to look at ray core doc to use AIR.

I'm fine with going with "Resource usage" for now, being a bit ambiguous before we add the related doc into AIR.

@richardliaw Any thoughts here on how we could better fill the gap?

File: python/ray/tune/utils/resource_updater.py [hunk 1]
```diff
@@ -290,39 +290,35 @@ def update_avail_resources(self, num_retries=5):
...
-                    total_resources.pop("CPU", 0),
-                    self._avail_resources.cpu,
-                    total_resources.pop("GPU", 0),
-                    self._avail_resources.gpu,
-                    _to_gb(total_resources.pop("memory", 0.0)),
-                    _to_gb(self._avail_resources.memory),
-                    _to_gb(total_resources.pop("object_store_memory", 0.0)),
-                    _to_gb(self._avail_resources.object_store_memory),
-                )
+            status = "Logical Resource Usage: {}/{} CPUs, {}/{} GPUs".format(
```
Comment (krfricke) [inline python/ray/tune/utils/resource_updater.py:296]:
```suggestion
            status = "Logical resource usage: {}/{} CPUs, {}/{} GPUs".format(
```

File: python/ray/tune/utils/resource_updater.py [hunk 2]
```diff
@@ -290,39 +290,35 @@ def update_avail_resources(self, num_retries=5):
...
+                        or "_group_" not in name
+                    )
                 ]
             )
             if customs:
                 status += f" ({customs})"
             return status
         else:
-            return "Resources requested: ?"
+            return "Logical Resource Usage: ?"
```
Comment (krfricke) [inline python/ray/tune/utils/resource_updater.py:321]:
```suggestion
            return "Logical resource usage: ?"
```

**Review by krfricke — APPROVED** [review]
Logical resource usage is fine with me if we consistently use this term and users understand it.

**Pull request merged**